libs/wingsdk/src/target-sim/queue.inflight.ts

```diff
@@ -139,6 +154,18 @@ export class Queue implements IQueueClient, ISimulatorResourceInstance {
   }
 }
 
+class QueueMessage {
+  retentionTimeout: Date;
+  payload: string;
+
+  constructor(retentionPeriod: number, message: string) {
+    const currentTime = new Date();
+    currentTime.setSeconds(retentionPeriod);
```

Comment:
I believe if we dont capture the current seconds from time when you call `.setSeconds()` there is a bug here. IE if the current time was 
`13:29:56.245Z` and the retention period came in as two minutes (120) then the time will be set to ``13:31:00.245Z` instead of `13:31:56.245Z` so its more like the retention period is 64 seconds.
```suggestion
    currentTime.setSeconds(retentionPeriod + currentTime.getSeconds());
```